src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -1674,13 +1675,22 @@ oop JavaThread::current_park_blocker() {
 }
 
 // Print stack trace for checked JNI warnings and JNI fatal errors.
-// This is the external format from above, but selecting the platform
-// or vthread as applicable.
+// This is the external format, selecting the platform or vthread
+// as applicable, and allowing for a native-only stack.
 void JavaThread::print_jni_stack() {
-  if (is_vthread_mounted()) {
-    print_vthread_stack_on(tty);
+  if (!has_last_Java_frame()) {
+    assert(this == JavaThread::current(), "Can't print native stack of other threads");
+    ResourceMark rm(this);
+    char* buf = NEW_RESOURCE_ARRAY_RETURN_NULL(char, O_BUFLEN);
+    if (buf == nullptr) {
+      tty->print_cr("Unable to print native stack - out of memory");
+      return;
+    }
+    frame f = os::current_frame();
```

Comment:
Is the function `JavaThread::print_jni_stack()` to print current JavaThread only or it was initially designed to print any JavaThread?

Reply:
It is only used for the current thread. That is a mistake on my part making it look like it works for any thread - it potentially could but only if the target was made safe.

Good catch!